### PR TITLE
feat: add overwrite confirmation when genesis file is existed

### PR DESCRIPTION
**Pull request by Halimao**
## Overview

This PR add support for displaying cover confirmation prompt to user when genesis file is existed

- **New Features**
  - Addition of displaying overwrite confirmation prompt to user when genesis file is existed
  - Addition of a new bool flag `yes`, used to Skip overwrite confirmation prompt when genesis.json is existed(in order to compatible with some old scripts)

----
## Screenshots

image

### CODE REVIEW

File: cmd/celestia-appd/cmd/download_genesis.go [hunk 1]
```diff
@@ -63,6 +82,7 @@ func downloadGenesisCommand() *cobra.Command {
 			return nil
 		},
 	}
+	cmd.Flags().BoolP(flagYes, "y", false, "Skip overwrite confirmation prompt when genesis.json is existed")
```
Comment (rootulp) [inline cmd/celestia-appd/cmd/download_genesis.go:85]: [nit] grammar 

```suggestion
	cmd.Flags().BoolP(flagYes, "y", false, "Skip overwrite confirmation prompt when genesis.json already exists")
```

File: cmd/celestia-appd/cmd/download_genesis.go [hunk 2]
```diff
@@ -35,6 +41,19 @@ func downloadGenesisCommand() *cobra.Command {
 				return fmt.Errorf("unknown chain-id: %s. Must be: celestia, mocha-4, or arabica-10", chainID)
 			}
 			outputFile := server.GetServerContextFromCmd(cmd).Config.GenesisFile()
+			// confirm overwrite, unless -y is passed
```
Comment (rootulp) [inline cmd/celestia-appd/cmd/download_genesis.go:44]:
```suggestion
			// confirm overwrite if --yes is not provided
```

File: cmd/celestia-appd/cmd/download_genesis.go [hunk 3]
```diff
@@ -35,6 +41,19 @@ func downloadGenesisCommand() *cobra.Command {
...
+			if skip, _ := cmd.Flags().GetBool(flagYes); !skip {
+				// if file is existed
```
Comment (rootulp) [inline cmd/celestia-appd/cmd/download_genesis.go:46]:
```suggestion
				// if file already exists
```

**Review by rootulp — COMMENTED** [review]
Thanks for the contribution! Is this feature something you want or has someone else requested it? 

Is it possible to refactor the implementation to reduce the amount of indentation?

**Comment by Halimao** [comment]
> Thanks for the contribution! Is this feature something you want or has someone else requested it?
> 
> Is it possible to refactor the implementation to reduce the amount of indentation?

I ran `celestia-appd download-genesis` for testing, and found
`celestia-appd download-genesis celestia` or 
`celestia-appd download-genesis mocha-4` or 
`celestia-appd download-genesis  arabica-10`
 
will use the same path `~/.celestia-app/config/genesis.json`. So I think it would be better to prompt user for this confirmation.

**Comment by Halimao** [comment]
@rootulp Hi sir, just tried and didn't find the way to reduce the amount of indentation, like check file stat `_, err := os.Stat(outputFile)` no matter `--yes` is provided, but this will lead to extra io.

Could u please give me some suggestions😀

**Review by evan-forbes — COMMENTED** [review]
imo, calling the download function is inherently giving permission to overwite the existing genesis, so we should strive to keep the UX clean and just always overwrite

**Comment by rootulp** [comment]
Going to close this based on Evan's last comment.

Note in response to @Halimao's https://github.com/celestiaorg/celestia-app/pull/2967#issuecomment-1870836008, I think a user could work around this by specifying a home directory:

```shell
$ celestia-appd download-genesis mocha-4 --home ~/.celestia-app-mocha
Downloading genesis file for mocha-4 to /Users/rootulp/.celestia-app-mocha/config/genesis.json
Downloaded genesis file for mocha-4 to /Users/rootulp/.celestia-app-mocha/config/genesis.json
SHA-256 hash verified for mocha-4

$ celestia-appd download-genesis arabica-10 --home ~/.celestia-app-arabica
Downloading genesis file for arabica-10 to /Users/rootulp/.celestia-app-arabica/config/genesis.json
Downloaded genesis file for arabica-10 to /Users/rootulp/.celestia-app-arabica/config/genesis.json
SHA-256 hash verified for arabica-10
```

**Pull request closed without merging**